scripts/Dockerfile

```
@@ -21,17 +21,20 @@ ENV CMAKE_CXX_COMPILER=/usr/bin/g++
 ENV CMAKE_MAKE_PROGRAM=/usr/bin/make
 ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib
 
-RUN conda create -n pybamm python=3.11
-RUN conda init --all
-SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
-RUN conda install -y pip
+# Migrating to venv and create a virtual environment for python 3.12
+ENV VIRTUAL_ENV=/home/pybamm/venv
+RUN python3.12 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
-RUN pip install --upgrade --user pip setuptools wheel wget
-RUN pip install cmake
+RUN pip install uv
+RUN uv pip install --upgrade setuptools wheel wget && \
+uv pip install cmake
+RUN #!/bin/bash && \
+source /home/pybamm/venv/bin/activate;
```

Can we just do it this way to make it look a bit simpler?
```suggestion
RUN pip install uv
RUN uv pip install --upgrade setuptools wheel wget cmake
RUN #!/bin/bash && source /home/pybamm/venv/bin/activate;
```